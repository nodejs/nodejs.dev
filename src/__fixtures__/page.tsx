import {
  PaginationInfo,
  LearnTemplateContext,
  NavigationData,
  PostTemplateData,
  PostTemplateContext,
  NodeReleaseData,
  PageTableOfContents,
  BlogCategory,
  BlogPost,
  ArticleData,
} from '../types';
import mockMDXBodyContent from './mockMDXBodyContent';

export const mockTableOfContents: PageTableOfContents = {
  items: [
    { title: 'mock-title', url: '#mock-title' },
    { title: 'mock-title-2', url: '#mock-title-2' },
  ],
};

export const createPaginationInfo = (): PaginationInfo =>
  ({
    slug: 'test-slug',
    title: 'test-title',
  } as PaginationInfo);

export const createNavigationSectionData = (): NavigationData =>
  ({
    'test-section': [
      {
        slug: 'test-slug-1',
        title: 'test-title-1',
      },
      {
        title: 'test-title-2',
        slug: 'test-slug-2',
      },
    ],
    'test-section2': [
      {
        slug: 'test-slug-3',
        title: 'test-title-3',
      },
      {
        title: 'test-title-4',
        slug: 'test-slug-4',
      },
    ],
  } as NavigationData);

export const createLearnPageData = () => ({
  data: {
    articleCurrentLanguage: {
      body: mockMDXBodyContent,
      tableOfContents: mockTableOfContents,
      frontmatter: {
        title: 'test-title',
        description: 'test-description',
      },
      fields: {
        authors: ['test-user1', 'test-user2'],
      },
    } as ArticleData,
  },
});

export const createNodeReleasesData = (): NodeReleaseData[] =>
  [
    {
      endOfLife: '2022-04-30',
      ltsStart: '2019-10-21',
      maintenanceStart: '2020-11-30',
      initialRelease: '2019-04-23',
      codename: 'erbium',
      version: 'v12',
      fullVersion: 'v12.0.0',
      status: 'Maintenance LTS',
      isLts: false,
    },
    {
      endOfLife: '2023-04-30',
      ltsStart: '2020-10-17',
      maintenanceStart: '2021-10-19',
      initialRelease: '2020-04-21',
      codename: 'fermium',
      version: 'v14',
      fullVersion: 'v14.0.0',
      status: 'Active LTS',
      isLts: true,
    },
    {
      endOfLife: '2025-04-30',
      ltsStart: '2022-10-25',
      maintenanceStart: '2023-10-18',
      initialRelease: '2022-04-19',
      codename: 'v18',
      version: 'v18',
      fullVersion: 'v18.0.0',
      status: 'Current',
      isLts: false,
    },
  ] as NodeReleaseData[];

export const createLearnPageContext = (): LearnTemplateContext =>
  ({
    slug: 'test-slug',
    relativePath: 'test-path',
    next: createPaginationInfo(),
    previous: createPaginationInfo(),
    navigationData: createNavigationSectionData(),
  } as LearnTemplateContext);

export const createBlogPageContext = (): PostTemplateContext => ({
  slug: 'test-slug',
  relativePath: 'test-path',
  next: createPaginationInfo(),
  previous: createPaginationInfo(),
  navigationData: createNavigationSectionData(),
  recent: [
    {
      node: {
        frontmatter: {
          title: 'title-mock',
          category: {
            name: 'category-mock',
            slug: 'category-mock-slug',
          },
          blogAuthors: [
            {
              id: 'id-mock',
              name: 'name-mock',
              website: 'url-mock',
            },
          ],
        },
        fields: {
          date: '11/11/2022',
          slug: 'slug-mock',
          readingTime: { text: 'text-mock' },
        },
      },
    },
  ],
});

export const createBlogData = (): {
  posts: { edges: BlogPost[] };
  categories: { edges: BlogCategory[] };
} => ({
  posts: {
    edges: [
      {
        node: {
          frontmatter: {
            title: 'Mock blog title',
            category: {
              name: 'mock-category',
              slug: 'Mock Category Slug',
            },
            blogAuthors: [
              {
                name: 'Mock author name',
                website: 'Mock URL',
              },
            ],
          },
          fields: {
            date: '11/11/2022',
            slug: 'Mock blog slug',
            readingTime: {
              text: '1 min read',
            },
          },
        },
      },
    ],
  },
  categories: {
    edges: [],
  },
});

export const createBlogPageData = (): PostTemplateData => ({
  mdx: {
    body: mockMDXBodyContent,
    excerpt: 'excerpt-mock',
    frontmatter: {
      title: 't,itle-mock',
      blogAuthors: [
        {
          id: 'id-mock',
          name: 'name-mock',
          website: 'url-mock',
        },
      ],
    },
    fields: { slug: 'slug-mock', date: '11/11/2022' },
  },
});

export const createGeneralPageData = () => ({
  data: {
    articleCurrentLanguage: {
      fields: {
        authors: ['author-mock'],
      },
      frontmatter: {
        description: 'Mock Description',
        title: 'Mock Title',
      },
      body: mockMDXBodyContent,
      tableOfContents: mockTableOfContents,
    } as ArticleData,
  },
});

export const createResourcesData = () => ({
  data: {
    articleCurrentLanguage: {
      fields: {
        authors: ['MrJithil'],
      },
      frontmatter: {
        description: 'Mock Description',
        title: 'Mock Title',
      },
      body: mockMDXBodyContent,
      tableOfContents: mockTableOfContents,
    } as ArticleData,
  },
});

export const createPrivacyData = () => ({
  data: {
    articleCurrentLanguage: {
      fields: {
        authors: ['Author'],
      },
      frontmatter: {
        description: 'Description',
        title: 'Title',
      },
      body: mockMDXBodyContent,
      tableOfContents: mockTableOfContents,
    } as ArticleData,
  },
});
