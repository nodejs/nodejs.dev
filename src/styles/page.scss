.page > .article-reader {
  margin: 0 auto;
}
