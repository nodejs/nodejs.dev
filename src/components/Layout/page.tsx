import React from 'react';
import Article from '../Article';
import Layout from '.';

import '../../styles/page.scss';

interface Page {
  editPath?: string;
  html: string;
  tableOfContents: string;
  title: string;
  description: string;
  authors: string[];
  location?: Location;
}

export default function PageLayout(props: Page): JSX.Element {
  const {
    editPath,
    html,
    tableOfContents,
    title,
    description,
    authors,
    location,
  } = props;
  return (
    <Layout title={title} location={location} description={description}>
      <main className="page">
        <Article
          html={html}
          title={title}
          authors={authors}
          editPath={editPath}
          tableOfContents={tableOfContents}
        />
      </main>
    </Layout>
  );
}
