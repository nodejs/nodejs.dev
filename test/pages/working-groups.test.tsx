import React from 'react';
import { render } from '@testing-library/react';
import WorkingGroupsPage from '../../src/pages/working-groups';
import '../__mocks__/intersectionObserverMock';
import { createGeneralPageData } from '../__fixtures__/page';

const mockData = createGeneralPageData();

describe('Working Groups page', () => {
  it('renders correctly', () => {
    const { container } = render(<WorkingGroupsPage data={mockData.data} />);
    expect(container).toMatchSnapshot();
  });
});
