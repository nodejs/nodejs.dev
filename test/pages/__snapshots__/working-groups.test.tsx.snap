// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Working Groups page renders correctly 1`] = `
<div>
  <div
    class="layout-container"
  >
    <nav
      aria-label="Primary"
      class="nav"
    >
      <div
        class="nav__container"
      >
        <div
          class="nav__startwrapper"
        >
          <a
            href="/"
          >
            <div
              class="logo"
            >
              <svg
                class="nav__logo light-mode-only"
              />
              <svg
                class="nav__logo dark-mode-only"
              />
            </div>
          </a>
        </div>
        <ul
          class="nav__tabs__container"
        >
          <li
            class="nav__tabs"
          >
            <a
              class="activeStyleTab"
              href="/learn"
            >
              Learn
            </a>
          </li>
          <li
            class="nav__tabs"
          >
            <a
              class="activeStyleTab"
              href="https://nodejs.org/en/docs/"
              rel="noopener noreferrer"
              target="_blank"
            >
              Documentation
            </a>
          </li>
          <li
            class="nav__tabs"
          >
            <a
              class="activeStyleTab"
              href="/download"
            >
              Download
            </a>
          </li>
          <li
            class="nav__tabs"
          >
            <a
              class="activeStyleTab"
              href="/community"
            >
              Community
            </a>
          </li>
        </ul>
        <div
          class="nav__endwrapper"
        >
          <ul
            class="right-container"
          >
            <li
              class="nav__tabs"
            >
              <div
                class="searchBarContainer"
                style="min-height: 3em; width: 100%; max-width: 7em; box-shadow: none;"
              >
                <div
                  class="searchInputContainer"
                >
                  <i
                    class="material-icons searchIcon"
                  >
                    search
                  </i>
                  <input
                    autocomplete="off"
                    class="inputText"
                    name="query"
                    placeholder="Search"
                    value=""
                  />
                </div>
              </div>
            </li>
            <li
              class="nav__tabs nav__tabs--right"
            >
              <button
                class="dark-mode-toggle"
                type="button"
              >
                <span
                  class="sr-only"
                >
                  Toggle Dark Mode
                </span>
                <i
                  class="material-icons light-mode-only theme-buttons"
                >
                  nights_stay
                </i>
                <i
                  class="material-icons dark-mode-only theme-buttons"
                >
                  wb_sunny
                </i>
              </button>
            </li>
            <li
              class="nav__tabs"
            >
              <a
                href="https://github.com/nodejs/nodejs.dev"
                rel="noopener noreferrer"
                target="_blank"
              >
                <span
                  class="sr-only"
                >
                  GitHub
                </span>
                <svg
                  fill="var(--color-text-accent)"
                  style="margin-bottom: -3px;"
                />
              </a>
            </li>
          </ul>
        </div>
      </div>
    </nav>
    <main
      class="grid-container"
    >
      <nav
        class="side-nav-about"
      >
        <button
          class="side-nav__open"
          type="button"
        >
          Menu
        </button>
        <ul
          class="community-nav__list"
        >
          <a
            class="t-body2 side-nav__item-community"
            href="/about"
            id="link-about"
          >
            About
          </a>
          <a
            class="t-body2 side-nav__item-community"
            href="/community"
            id="link-community"
          >
            Community
          </a>
          <a
            class="t-body2 side-nav__item-community"
            href="/download/package-manager"
            id="link-download/package-manager"
          >
            Package Manager
          </a>
          <a
            class="t-body2 side-nav__item-community"
            href="/privacy"
            id="link-privacy"
          >
            Privacy Policy
          </a>
          <a
            class="t-body2 side-nav__item-community"
            href="/governance"
            id="link-governance"
          >
            Project Governance
          </a>
          <a
            class="t-body2 side-nav__item-community"
            href="/releases"
            id="link-releases"
          >
            Releases
          </a>
          <a
            class="t-body2 side-nav__item-community"
            href="/resources"
            id="link-resources"
          >
            Resources
          </a>
          <a
            class="t-body2 side-nav__item-community"
            href="/security"
            id="link-security"
          >
            Security Reporting
          </a>
          <a
            class="t-body2 side-nav__item-community"
            href="/trademark"
            id="link-trademark"
          >
            Trademark Policy
          </a>
          <a
            class="t-body2 side-nav__item-community side-nav__item-community--active"
            href="/working-groups"
            id="link-working-groups"
          >
            Working Groups
          </a>
        </ul>
      </nav>
      <article
        class="article-reader"
      >
        <h1
          class="article-reader__headline"
        >
          Mock Title
        </h1>
        <div>
          <p>
            Node.js can be installed in different ways. This post highlights the most common and convenient ones.
          </p>
          <p>
            Official packages for all the major platforms are available at 
            <a
              href="https://nodejs.dev/download/"
            >
              https://nodejs.dev/download/
            </a>
            .
          </p>
          <p>
            One very convenient way to install Node.js is through a package manager. In this case, every operating system has its own.
          </p>
          <p>
            Other package managers for MacOS, Linux, and Windows are listed in 
            <a
              href="https://nodejs.dev/download/package-manager/"
            >
              https://nodejs.dev/download/package-manager/
            </a>
          </p>
          <p>
            <code
              class="language-text"
            >
              nvm
            </code>
             is a popular way to run Node.js. It allows you to easily switch the Node.js version, and install new versions to try and easily rollback if something breaks, for example.
          </p>
          <p>
            It is also very useful to test your code with old Node.js versions.
          </p>
          <p>
            See 
            <a
              href="https://github.com/nvm-sh/nvm"
            >
              https://github.com/nvm-sh/nvm
            </a>
             for more information about this option.
          </p>
          <p>
            In any case, when Node.js is installed you'll have access to the 
            <code
              class="language-text"
            >
              node
            </code>
             executable program in the command line.
          </p>
        </div>
        <ul
          class="list"
        >
          <h5>
            Contributors
          </h5>
          <li>
            <a
              class="link"
              href="https://github.com/author-mock"
              rel="noopener noreferrer"
              style="margin-left: 0px;"
              target="_blank"
              title="author-mock"
            >
              <img
                alt="author-mock"
                class="img"
                src="https://github.com/author-mock.png?size=60"
              />
            </a>
          </li>
        </ul>
        <div
          class="edit"
        >
          <a
            class="link"
            href="https://github.com/nodejs/nodejs.dev/edit/main/content/about/working-groups.md"
          >
            <span>
              Edit this page on GitHub
            </span>
             
            <svg
              class="icon"
              fill="currentColor"
              height="1em"
              viewBox="0 0 40 40"
              width="1em"
            >
              <path
                d="m34.5 11.7l-3 3.1-6.3-6.3 3.1-3q0.5-0.5 1.2-0.5t1.1 0.5l3.9 3.9q0.5 0.4 0.5 1.1t-0.5 1.2z m-29.5 17.1l18.4-18.5 6.3 6.3-18.4 18.4h-6.3v-6.2z"
              />
            </svg>
          </a>
        </div>
      </article>
    </main>
    <section
      class="bottom-info"
    >
      <div
        class="random-contributor"
      />
    </section>
    <footer
      class="footer"
    >
      <ul
        class="footer__left"
      >
        <li>
          <a
            class="footer__link"
            href="/trademark"
          >
            Trademark Policy
          </a>
        </li>
        <li>
          <a
            class="footer__link"
            href="/privacy"
          >
            Privacy Policy
          </a>
        </li>
        <li>
          <a
            class="footer__link"
            href="https://github.com/openjs-foundation/cross-project-council/blob/main/CODE_OF_CONDUCT.md#contributor-covenant-code-of-conduct"
          >
            Code of Conduct
          </a>
        </li>
        <li>
          <a
            class="footer__link"
            href="/security"
          >
            Security Reporting
          </a>
        </li>
        <li>
          <a
            class="footer__link"
            href="/about"
          >
            About
          </a>
        </li>
        <li>
          <a
            class="footer__link"
            href="https://nodejs.org/en/blog/"
          >
            Blog
          </a>
        </li>
      </ul>
      <ul
        class="footer__right"
      >
        <li>
          © OpenJS Foundation
        </li>
        <li>
          <a
            aria-label="Node.js Github Page Link"
            href="https://github.com/nodejs/node"
            rel="noopener noreferrer"
            target="_blank"
          >
            <span
              class="sr-only"
            >
              GitHub
            </span>
            <svg
              fill="var(--color-text-secondary)"
            />
          </a>
        </li>
        <li>
          <a
            aria-label="Node.js Twitter Link"
            href="https://twitter.com/nodejs"
            rel="noopener noreferrer"
            target="_blank"
          >
            <svg
              fill="var(--color-text-secondary)"
            />
          </a>
        </li>
        <li>
          <a
            aria-label="Node.js Slack Link"
            href="https://slack.openjsf.org"
            rel="noopener noreferrer"
            target="_blank"
          >
            <svg
              fill="var(--color-text-secondary)"
            />
          </a>
        </li>
      </ul>
    </footer>
  </div>
</div>
`;
